mlrun/config.py

```diff
@@ -553,6 +556,18 @@ def get_valid_function_priority_class_names():
     def is_running_on_iguazio() -> bool:
         return config.igz_version is not None and config.igz_version != ""
 
+    @staticmethod
+    def get_security_context_enrichment_group_id(user_unix_id: int) -> int:
+        enrichment_group_id = int(
+            config.function.spec.security_context.enrichment_group_id
+        )
+
+        # if enrichment group id is -1 we set group id to user unix id
+        enrichment_group_id = (
+            enrichment_group_id if enrichment_group_id != -1 else user_unix_id
+        )
```

Comment:
Raise an exception if `enrichment group == -1` and `user_unix_id` is not defined